[Brendan8899] iP

Brendan8899 opened the pull request:
# Alpha ChatBot

Alpha ChatBot frees your mind from having to remember things you need to do. It's:

- **Text-based** 📝
- **Easy to learn** 🧠
- ~FAST~ **SUPER FAST** 🚀 to use

> "The Journey of a Thousand Miles Begins with a Single Step" - Lao Zi

## How to Use Alpha ChatBot

All you need to do is:

1. **Download it** from [here](https://github.com/Brendan8899/ip).
2. More instructions coming soon!

## Features

- [x] **Managing Tasks**: Add, delete, and modify tasks easily.
- [x] **Reminders**: Get reminders for upcoming tasks.
- [x] **Automatic Reminders**: Alpha automatically reminds you about tasks due soon.
- [x] **Intelligent Search Function**: Find your tasks quickly using keywords.

## Getting Started

### Prerequisites

Make sure you have the following software installed:

- Java 17 or later. You can download it [here](https://www.oracle.com/java/technologies/javase-jdk17-downloads.html).

If you are a Java programmer, you can use it to practice Java too. Here's the code that generates the reminders and welcome message whenever the user opens the application

```ruby
  public void setAlpha(Alpha d) {
        alpha = d;
        Ui ui = new Ui();
        
        String welcomeMessage = ui.welcomeMessage();
        String reminderMessages = d.taskReminder();
        
        dialogContainer.getChildren().addAll(
                DialogBox.getDukeDialog(welcomeMessage, alphaImage),
                DialogBox.getDukeDialog(reminderMessages, alphaImage)
        );
    } 
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Brendan8899 committed:
Rename main class to alpha

Brendan8899 committed:
Add greeting

Brendan8899 committed:
Add Scanner to take in user input

Brendan8899 committed:
Implement storage for echo

Brendan8899 committed:
Create add and list features

Brendan8899 committed:
Fix faulty bye response

Brendan8899 committed:
Implement Mark as Done status

Brendan8899 committed:
Immplement ToDos, Events and Deadlines

Brendan8899 committed:
Immplement ToDos, Events and Deadlines

Brendan8899 committed:
Fix identation for code

Brendan8899 committed:
Plan Test Cases Input and Expected Output

Brendan8899 committed:
Handle all exceptions for invalid input format

Brendan8899 committed:
Add Delete Function

Brendan8899 committed:
Implemented Enum for Possible Commands

Brendan8899 committed:
Fix Bug for Todo Input

Brendan8899 committed:
Implement save function in System Memory

Brendan8899 committed:
Merge branch 'branch-Level-7'

Brendan8899 committed:
Implement LocalDate and formatting for dates in Task List

Brendan8899 committed:
Merge branch 'branch-Level-8'

Brendan8899 committed:
Implement A More OOP Restructure

Brendan8899 committed:
Merge branch 'A_More_OOP_Feature_Branch'

Brendan8899 committed:
Add Package for all classes

Brendan8899 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Brendan8899 committed:
Implement Gradle Function to App

Brendan8899 committed:
Add JUnit Test for Parser Functions

Brendan8899 committed:
Add 1 more test case

Brendan8899 committed:
Implement creation of JAR File

Brendan8899 committed:
Add header comments to non-private methods and classes

Brendan8899 committed:
Merge branch 'A-JavaDoc'

Brendan8899 committed:
Follow Coding Standards for Java

Brendan8899 committed:
Fix directory case issue and remove incorrect Utility directory

Brendan8899 committed:
Implement Coding Standards

Brendan8899 committed:
Fix Indentation

Brendan8899 committed:
Implement Search Parameter Regex

Brendan8899 committed:
Add Java Doc Comments for new methods

Brendan8899 committed:
Merge branch 'Level-9'

rollingpencil commented on src/main/java/tasktypes/Task.java:
```diff
@@ -0,0 +1,100 @@
+     *
+     * @return a string in the format "| doneStatus | description"
+     */
+    public String storageFormat() {
```
```suggestion
    public String toStorageFormatString() {
```
`storageFormat` as a variable name is vague in terms of describing what the method does. Perhaps `toStringFormatString` might be more descriptive?

rollingpencil commented on src/main/java/utility/TaskList.java:
```diff
@@ -0,0 +1,153 @@
+     * @param searchParameter the keyword or phrase to search for within the task descriptions
+     * @return an {@code ArrayList} of tasks that match the search criteria
+     */
+    public ArrayList<Task> findLists(String searchParameter) {
```
```suggestion
    public ArrayList<Task> findItemsInList(String searchParameter) {
```
`findLists` might be misleading as it describes there exists multiple lists. Perhaps `findItemsInList` might be more suitable?

rollingpencil commented on src/main/java/utility/Ui.java:
```diff
@@ -0,0 +1,123 @@
+    /**
+     * Displays the welcome message to the user when the application starts.
+     */
+    public void welcomeMessage() {
```
```suggestion
    public void showWelcomeMessage() {
```
`welcomeMessage` might not be very descriptive of the action this message is going to perform. Perhaps, adding a `show` or `display` prefix might better describe the method.

rollingpencil commented on src/test/java/tasktypes/DeadlineTest.java:
```diff
@@ -0,0 +1,13 @@
+
+public class DeadlineTest {
+    @Test
+    public void descriptionTest() {
```
```suggestion
    public void todoDescription_validDescription_success() {
```
`descriptionTest` is vague in terms of describing the expected input and result the test case should produce. Perhaps `todoDescription_validDescription_success` might be better?

rollingpencil commented on src/test/java/utility/ParserTest.java:
```diff
@@ -0,0 +1,31 @@
+public class ParserTest {
+    
+    @Test
+    public void extractCommandTest() {
```
```suggestion
    public void parserExtractCommand_validCommand_success() {
```
`extractCommandTest` is vague in terms of describing the expected input and result the test case should produce. Perhaps `parserExtractCommand_validCommand_success` might be better?

rollingpencil commented on src/test/java/utility/ParserTest.java:
```diff
@@ -0,0 +1,31 @@
+    }
+    
+    @Test
+    public void extractDescriptionTest() {
```
```suggestion
    public void parserExtractDescription_validCommand_success() {
```

rollingpencil commented on src/test/java/utility/ParserTest.java:
```diff
@@ -0,0 +1,31 @@
+    }
+    
+    @Test
+    public void extractFirstDateTest() {
```
```suggestion
    public void parserExtractFirstDate_validCommand_success() {
```

rollingpencil commented on src/test/java/utility/ParserTest.java:
```diff
@@ -0,0 +1,31 @@
+    }
+    
+    @Test
+    public void extractSecondDateTest() {
```
```suggestion
    public void parserExtractSecondDate_validCommand_success() {
```

rollingpencil reviewed (commented):
Looks good to me aside from a few nits to pick. Majority of the comments came from the naming of the test cases. Other than that, some method names could be more descriptive. Overall good job!

accountexeregister commented on src/main/java/alpha/Alpha.java:
```diff
@@ -0,0 +1,170 @@
+                ui.listTask(tasks);
+            } else if (inputCommand.equalsIgnoreCase(Commands.FIND.getCommand())) {
+                String searchParam = input.split(" ")[1];
+                ArrayList<Task> searchResult = tasks.findLists(searchParam);
```
Should `searchResult` be plural, since it is a list of tasks?

accountexeregister commented on src/main/java/alpha/Alpha.java:
```diff
@@ -0,0 +1,170 @@
+                ui.searchTask(searchResult);
+            } else if (inputCommand.equalsIgnoreCase(Commands.MARK.getCommand())) {
+                try {
+                    Integer indexInvolved = Integer.valueOf(input.split(" ")[1]);
```
Perhaps you can have a better variable name for `indexInvolved`?

accountexeregister commented on src/main/java/exception/AlphaException.java:
Does it make sense to have an Exception subclass that functions like an Exception? Would it be better to have more Exception subclasses that are more specific?

accountexeregister commented on src/main/java/tasktypes/Task.java:
```diff
@@ -0,0 +1,100 @@
+     * @param dateConversion the {@code LocalDate} to format
+     * @return a formatted date string in the "MMM d yyyy" pattern
+     */
+    public String dateFormatPrintVersion(LocalDate dateConversion) {
```
This should probably be something like `getDateFormatPrintVersion`, since it returns a string. `dateFormatPrintVersion` might be interpreted as a void method

accountexeregister commented on src/main/java/utility/Commands.java:
Nice, good idea using commands as enums

accountexeregister commented on src/main/java/utility/Parser.java:
Great job abstracting these functions to parse commands

Brendan8899 committed:
Add GUI Features for the Application

Brendan8899 committed:
Committing changes before merge

Brendan8899 committed:
Merge branch 'Level-10'

Brendan8899 committed:
Implement Assertions in Parser Utility Class to Test for Bad Input

Brendan8899 committed:
Add Enumeration for TaskName for sorting operation

Create Enumeration containing the different task names so that future
logic for sorting would be easier.

Brendan8899 committed:
Add Javadoc Comments for Parser Functions

Create 3 New Function in Parser.java to abstract text processing
operations. Add JavaDoc Comments for all 3.

Brendan8899 committed:
Remove Excess Spacing for UI.java

The Indentation space should be 4 spaces instead of 8. I corrected this
issue for UI.java

Brendan8899 committed:
Implement Horizontal Scaling of Screens

Edit the FXML File to make sure that horizontal scaling works

Brendan8899 committed:
Implement Automatic Reminder System

For tasks that have less than 1 week before due date, the chatbot will
automatically provide reminders.

Brendan8899 committed:
Merge pull request #3 from Brendan8899/branch-A-Assertions

Branch a assertions

Brendan8899 committed:
Merge branch 'A-CodeQuality'

Brendan8899 committed:
Merge branch 'A-BetterGui'

Brendan8899 committed:
Merge branch 'B-Reminders'

Brendan8899 committed:
Fix Identation issue for TaskList.java

Brendan8899 committed:
Add Javadoc Comments in TaskList for clarity

Brendan8899 committed:
Add files via upload

Brendan8899 committed:
Update README.md

Brendan8899 committed:
Update README.md

Brendan8899 committed:
Add JavaDoc Comments and Edge cases for delete, mark and unmark commands

Brendan8899 committed:
Merge branch 'master' of https://github.com/Brendan8899/ip
